python/filtering/filter_item_widget.py

```diff
@@ -356,23 +318,17 @@ def value(self, value):
         """
 
         if not isinstance(value, six.string_types):
-            assert False, "Attempt to set non-string data to QLineEdit"
```

Comment:
Not sure why this check is being removed here, are we expecting other types of searching values ?

Reply:
I originally removed the assert to just ignore it because we were getting boolean values passed in some times when check states are changed; however, you're right in asking if we should ever expect other types being passed, because we shouldn't in the first place.

I added back the assert, and then removed the signal/slot connection that was incorrectly trying to set the text filter item widgets value to the action's check state. While I was making this change, I also improved how the filter item value is set and to avoid triggering menu refresh if the value being set is the same as the current value.